File: frontend/src/types/WorkoutSession.ts
export type WorkoutSession = {
    id: string;
    workoutName: string;
    exercises: ExerciseSessionData[];
    startTime: string;
}

export type ExerciseSessionData = {
    exerciseId: string;
    exerciseName?: string;
    exerciseImage?: string;
    sets: number;
    reps: number;
    weight: number;
}

File: backend/src/main/java/com/yuliiaskrypnyk/backend/model/session/WorkoutSession.java
package com.yuliiaskrypnyk.backend.model.session;

import lombok.Builder;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.List;

@Builder
@Document(collection = "workout_sessions")
public record WorkoutSession(
        @Id String id,
        String workoutName,
        LocalDateTime startTime,
        List<ExerciseSessionData> exercises
) {
}
